lib/charms/data_platform_libs/v0/upgrade.py

```diff
@@ -930,7 +945,6 @@ def on_upgrade_changed(self, event: EventBase) -> None:
             # on k8s only the upgrading unit will receive the upgrade-charm event
             # and deferring will prevent the upgrade stack from being popped
             logger.debug("Cluster in recovery, deferring...")
-            event.defer()
```

Comment:
The defer seems to be unnecessary, and causing a concurrency on `upgrade_granted` calls